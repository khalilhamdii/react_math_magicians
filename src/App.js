import "./App.css";
import PropTypes from "prop-types";
import Big from "big.js";

const App = () => {
  return (
    <div className="App">
      <header className="App-header">
        <h1>Math Magicians App</h1>
      </header>
    </div>
  );
};

export default App;
