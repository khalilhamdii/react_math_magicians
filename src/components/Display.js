/* eslint-disable no-unused-vars */
import React from 'react';
import PropTypes from 'prop-types';
import Big from 'big.js';

function Display(props) {
  const { result } = props;
  return <div>{result}</div>;
}
Display.propTypes = {
  result: PropTypes.string,
};
Display.defaultProps = { result: 0 };

export default Display;
