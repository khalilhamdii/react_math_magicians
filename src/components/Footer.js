import React from 'react';
import '../stylesheets/footer.css';

const Footer = () => (
  <div className="footer-basic mt-5" style={{ background: '#F5913E' }}>
    <footer>
      <ul className="list-inline" style={{ color: 'white' }}>
        <li className="list-inline-item">
          <a href="/">Home</a>
        </li>
        <li className="list-inline-item">
          <a href="/calculator">Calculator</a>
        </li>
        <li className="list-inline-item">
          <a href="/quote">Quote</a>
        </li>
      </ul>
      <p className="copyright" style={{ fontSize: '16px', color: 'var(--dark)' }}>
        Math Magicians by Khalil Hamdi Â© 2021
      </p>
    </footer>
  </div>
);

export default Footer;
