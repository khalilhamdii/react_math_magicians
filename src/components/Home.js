import React, { useEffect } from 'react';
/* eslint-disable arrow-parens */
const Home = () => {
  useEffect(() => {
    const navLinks = document.querySelectorAll('.nav-link');
    const element = document.getElementById('1');
    navLinks.forEach((item) => item.classList.remove('active'));
    element.classList.add('active');
  }, []);
  return (
    <div className="container">
      <h1 style={{ color: 'rgb(255,255,255)' }}>Welcome to our page!</h1>
      <p style={{ color: 'white', fontSize: '25px' }}>
        <br />
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt
        ut labore et dolore magna aliqua. Utenim ad minim veniam, quis nostrud exercitation ullamco
        laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in
        voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat
        cupidatat non proident, sunt inculpa qui o!cia deserunt mollit anim id est laborum.
      </p>
      <br />
      <p style={{ color: 'white', fontSize: '25px' }}>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt
        ut labore et dolore magna aliqua. Utenim ad minim veniam, quis nostrud exercitation ullamco
        laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in
        voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat
        cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum.
      </p>
      <br />
    </div>
  );
};

export default Home;
