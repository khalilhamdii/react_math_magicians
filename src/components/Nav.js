import React from 'react';
import '../stylesheets/style.css';

const Nav = () => (
  <nav
    className="navbar navbar-light navbar-expand-md navigation-clean mb-5"
    style={{ background: '#F5913E' }}
  >
    <div className="container">
      <a
        className="navbar-brand"
        href="/"
        style={{ fontFamily: 'Aclonica, sans-serif', fontSize: '30px' }}
      >
        Math Magicians
      </a>
      <button
        data-toggle="collapse"
        className="navbar-toggler"
        type="button"
        data-target="#navcol-1"
      >
        <span className="sr-only">Toggle navigation</span>
        <span className="navbar-toggler-icon" />
      </button>
      <div className="collapse navbar-collapse" id="navcol-1">
        <ul className="nav navbar-nav ml-auto font-weight-bold">
          <li className="nav-item">
            <a className="nav-link" href="/" style={{ color: 'rgb(255,255,255)' }}>
              HOME
            </a>
          </li>
          <li className="nav-item">
            <a className="nav-link" href="/calculator" style={{ color: 'rgb(255,255,255)' }}>
              CALCULATOR
            </a>
          </li>
          <li className="nav-item">
            <a className="nav-link" href="/quote" style={{ color: 'rgb(255,255,255)' }}>
              QUOTE
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
);

export default Nav;
