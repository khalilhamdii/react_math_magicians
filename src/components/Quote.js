import React from 'react';
import '../stylesheets/style.css';

const Quote = () => (
  <div className="container">
    <h1 style={{ color: 'rgb(255,255,255)' }}>Quote of the day</h1>
    <p
      style={{
        fontFamily: 'Quattrocento Sans, sans-serif',
        fontSize: '48px',
        color: 'white',
        textAlign: 'center',
      }}
    >
      <br />
      <br />
      &quot;Mathematics is not about numbers, equations, computations, or algorithms: it is about
      understanding.&quot; â€“ William Paul Thurston
      <br />
      <br />
      <br />
    </p>
  </div>
);

export default Quote;
