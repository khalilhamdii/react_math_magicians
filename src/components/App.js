/* eslint-disable react/no-unused-state */
/* eslint-disable arrow-parens */
import React, { useState } from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';

const App = () => {
  const [total, setTotal] = useState(null);
  const [next, setnext] = useState(null);
  const [operation, setoperation] = useState(null);
  const [log, setlog] = useState(null);

  const handleClick = (btnName) => {
    const data = {
      total,
      next,
      operation,
      log,
    };
    const modifiedData = calculate(data, btnName);
    setTotal(modifiedData.total);
    setnext(modifiedData.next);
    setoperation(modifiedData.operation);
    setlog(modifiedData.log);
  };
  return (
    <div className="container d-flex flex-row justify-content-between flex-wrap">
      <h1 className="flex-md-row" style={{ color: 'rgb(255,255,255)' }}>
        Let&apos;s do some math!
      </h1>
      <div className="col-12-md" style={{ width: '700px' }}>
        <Display result={next} log={log} />
        <ButtonPanel clickHandler={handleClick} />
      </div>
    </div>
  );
};

export default App;
