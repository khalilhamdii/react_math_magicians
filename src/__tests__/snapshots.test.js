import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from '../components/App';
import Display from '../components/Display';
import Nav from '../components/Nav';
import Footer from '../components/Footer';
import Home from '../components/Home';
import Quote from '../components/Quote';

describe('Snapshots', () => {
  beforeEach(() => render(<Nav />));

  it('renders App correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Home correctly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Quote correctly', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Display correctly', () => {
    const tree = renderer.create(<Display />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Nav correctly', () => {
    const tree = renderer.create(<Nav />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Footer correctly', () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
