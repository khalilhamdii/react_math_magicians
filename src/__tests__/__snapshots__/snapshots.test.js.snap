// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots renders App correctly 1`] = `
<div
  className="container d-flex flex-row justify-content-between flex-wrap"
>
  <h1
    className="flex-md-row"
    style={
      Object {
        "color": "rgb(255,255,255)",
      }
    }
  >
    Let's do some math!
  </h1>
  <div
    className="col-12-md"
    style={
      Object {
        "width": "700px",
      }
    }
  >
    <div
      style={
        Object {
          "background": "#858694",
          "color": "white",
          "height": "150px",
          "textAlign": "right",
        }
      }
    >
      <div
        style={
          Object {
            "fontSize": "25px",
            "height": "50px",
            "padding": "5px",
            "paddingLeft": "10px",
            "paddingRight": "10px",
          }
        }
      >
        
      </div>
      <div
        style={
          Object {
            "borderStyle": "none",
            "fontSize": "45px",
            "height": "100px",
            "padding": "10px",
          }
        }
      >
        0
      </div>
    </div>
    <div>
      <div>
        <button
          className="calc-btn w-25 num-btn"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="calc-btn w-25 num-btn"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="calc-btn w-25 num-btn"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="calc-btn w-25 op-btn"
          onClick={[Function]}
          type="button"
        >
          ÷
        </button>
      </div>
      <div>
        <button
          className="calc-btn w-25 num-btn"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="calc-btn w-25 num-btn"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="calc-btn w-25 num-btn"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="calc-btn w-25 op-btn"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </div>
      <div>
        <button
          className="calc-btn w-25 num-btn"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="calc-btn w-25 num-btn"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="calc-btn w-25 num-btn"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="calc-btn w-25 op-btn"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div>
        <button
          className="calc-btn w-25 num-btn"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="calc-btn w-25 num-btn"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="calc-btn w-25 num-btn"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="calc-btn w-25 op-btn"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div>
        <button
          className="calc-btn w-50 num-btn"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="calc-btn w-25 num-btn"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="calc-btn w-25 op-btn"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshots renders Display correctly 1`] = `
<div
  style={
    Object {
      "background": "#858694",
      "color": "white",
      "height": "150px",
      "textAlign": "right",
    }
  }
>
  <div
    style={
      Object {
        "fontSize": "25px",
        "height": "50px",
        "padding": "5px",
        "paddingLeft": "10px",
        "paddingRight": "10px",
      }
    }
  >
    
  </div>
  <div
    style={
      Object {
        "borderStyle": "none",
        "fontSize": "45px",
        "height": "100px",
        "padding": "10px",
      }
    }
  >
    0
  </div>
</div>
`;

exports[`Snapshots renders Footer correctly 1`] = `
<div
  className="footer-basic mt-5"
  style={
    Object {
      "background": "#F5913E",
    }
  }
>
  <footer>
    <ul
      className="list-inline"
      style={
        Object {
          "color": "white",
        }
      }
    >
      <li
        className="list-inline-item"
      >
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li
        className="list-inline-item"
      >
        <a
          href="/calculator"
        >
          Calculator
        </a>
      </li>
      <li
        className="list-inline-item"
      >
        <a
          href="/quote"
        >
          Quote
        </a>
      </li>
    </ul>
    <p
      className="copyright"
      style={
        Object {
          "color": "var(--dark)",
          "fontSize": "16px",
        }
      }
    >
      Math Magicians by Khalil Hamdi © 2021
    </p>
  </footer>
</div>
`;

exports[`Snapshots renders Home correctly 1`] = `
<div
  className="container"
>
  <h1
    style={
      Object {
        "color": "rgb(255,255,255)",
      }
    }
  >
    Welcome to our page!
  </h1>
  <p
    style={
      Object {
        "color": "white",
        "fontSize": "25px",
      }
    }
  >
    <br />
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Utenim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt inculpa qui o!cia deserunt mollit anim id est laborum.
  </p>
  <br />
  <p
    style={
      Object {
        "color": "white",
        "fontSize": "25px",
      }
    }
  >
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Utenim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum.
  </p>
  <br />
</div>
`;

exports[`Snapshots renders Nav correctly 1`] = `
<nav
  className="navbar navbar-light navbar-expand-md navigation-clean mb-5"
  style={
    Object {
      "background": "#F5913E",
    }
  }
>
  <a
    className="navbar-brand"
    href="/"
    style={
      Object {
        "fontFamily": "Aclonica, sans-serif",
        "fontSize": "30px",
      }
    }
  >
    Math Magicians
  </a>
  <button
    className="navbar-toggler"
    data-target="#navcol-1"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="sr-only"
    >
      Toggle navigation
    </span>
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navcol-1"
  >
    <ul
      className="nav navbar-nav ml-auto font-weight-bold"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/"
          id="1"
        >
          HOME
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/calculator"
          id="2"
        >
          CALCULATOR
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/quote"
          id="3"
        >
          QUOTE
        </a>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`Snapshots renders Quote correctly 1`] = `
<div
  className="container"
>
  <h1
    style={
      Object {
        "color": "rgb(255,255,255)",
      }
    }
  >
    Quote of the day
  </h1>
  <p
    style={
      Object {
        "color": "white",
        "fontFamily": "Quattrocento Sans, sans-serif",
        "fontSize": "48px",
        "textAlign": "center",
      }
    }
  >
    <br />
    <br />
    "Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding." – William Paul Thurston
    <br />
    <br />
    <br />
  </p>
</div>
`;
